src/main/java/com/minecolonies/coremod/commands/AddOfficerCommand.java

```diff
@@ -0,0 +1,90 @@
+package com.minecolonies.coremod.commands;
+
+import com.minecolonies.coremod.colony.Colony;
+import com.minecolonies.coremod.colony.ColonyManager;
+import com.minecolonies.coremod.colony.permissions.Permissions;
+import net.minecraft.command.CommandException;
+import net.minecraft.command.ICommandSender;
+import net.minecraft.server.MinecraftServer;
+import net.minecraft.util.math.BlockPos;
+import net.minecraft.util.text.TextComponentString;
+import org.jetbrains.annotations.NotNull;
+
+import javax.annotation.Nullable;
+import java.util.Collections;
+import java.util.List;
+
+/**
+ * List all colonies.
+ */
+public class AddOfficerCommand extends AbstractSingleCommand
+{
+
+    public static final  String       DESC                            = "addOfficer";
+    private static final String       SUCCESS_MESSAGE             = "Succesfully added Player %s to colony %d";
+    private static final String       COLONY_NULL                     = "Couldn't find colony %d.";
+
+    /**
+     * Initialize this SubCommand with it's parents.
+     *
+     * @param parents an array of all the parents.
+     */
+    public AddOfficerCommand(@NotNull final String... parents)
+    {
+        super(parents);
+    }
+
+    @NotNull
+    @Override
+    public String getCommandUsage(@NotNull final ICommandSender sender)
+    {
+        return super.getCommandUsage(sender) + "<ColonyId> <(Optional)Player>";
+    }
+
+    @Override
+    public void execute(@NotNull final MinecraftServer server, @NotNull final ICommandSender sender, @NotNull final String... args) throws CommandException
+    {
+        final int colonyId = getIthArgument(args, 0, -1);
+
+        String playerName = null;
+
+        if (args.length >= 2)
+        {
+            playerName = args[1];
+        }
+
+        if(playerName == null || playerName.isEmpty())
+        {
+            playerName = sender.getName();
+        }
+
+        Colony colony = ColonyManager.getColony(colonyId);
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Local variable 'colony' could be declared final [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=pmd%3ALocalVariableCouldBeFinal)